tests/t286_agent_trigger.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+
+import subprocess as sp
+from time import sleep
+
+from runtest import TestBase
+
+class TestCase(TestBase):
+    def __init__(self):
+        TestBase.__init__(self, 'agent', """
+# DURATION     TID     FUNCTION
+            [  6999] | main() {
+            [  6999] |   func() {
+            [  6999] |     a() {
+            [  6999] |       b() {
+   1.056 ms [  6999] |         c();
+   2.113 ms [  6999] |       } /* b */
+   3.175 ms [  6999] |     } /* a */
+   3.175 ms [  6999] |   } /* func */
+            [  6999] |   func() {
+            [  6999] |     a() {
+   2.116 ms [  6999] |       b();
+   3.181 ms [  6999] |     } /* a */
+   3.182 ms [  6999] |   } /* func */
+            [  6999] |   func() {
+            [  6999] |     a() {
+            [  6999] |       b() {
+   1.057 ms [  6999] |         c();
+   2.116 ms [  6999] |       } /* b */
+   3.178 ms [  6999] |     } /* a */
+   3.179 ms [  6999] |   } /* func */
+            [  6999] |   func() {
+   3.175 ms [  6999] |     a();
+   3.177 ms [  6999] |   } /* func */
+            [  6999] |   func() {
+            [  6999] |     a() {
+            [  6999] |       b() {
+   1.057 ms [  6999] |         c();
+   2.119 ms [  6999] |       } /* b */
+   3.185 ms [  6999] |     } /* a */
+   3.185 ms [  6999] |   } /* func */
+  55.835 ms [  6999] | } /* main */
+""")
+
+
+    def client_send_command(self, pid, option):
+        self.subcmd = 'live'
+        self.option = '-p %d %s' % (pid, option)
+        self.exearg = ''
+        client_cmd = self.runcmd()
+        self.pr_debug('prerun command: ' + client_cmd)
+        client_p = sp.run(client_cmd.split())
+        return client_p.returncode
+
+    def prerun(self, timeout):
+        self.subcmd = 'record'
+        self.option  = '--keep-pid'
+        self.option += ' --no-libcall'
+        self.option += ' --agent'
+        self.exearg  = f't-{self.name}'
+        self.exearg += f' --delay'
+        record_cmd  = self.runcmd()
+        self.pr_debug("prerun command: " + record_cmd)
+        # bufsize=0 to write characters one by one
+        record_p = sp.Popen(record_cmd.split(), stdin=sp.PIPE, stderr=sp.PIPE, bufsize=0)
+
+        sleep(.05)              # time for the agent to start
+
+        if self.client_send_command(record_p.pid, '-T a@depth=2') != 0:
+            return TestBase.TEST_NONZERO_RETURN
+        record_p.stdin.write(b'0')
+
+        if self.client_send_command(record_p.pid, '-T a@depth,clear') != 0:
```

Comment:
Just curious, `-T a@clear,depth` should also work, right?

Reply:
Yes, the parsers are independent. As long as `clear` is found, it reverses the flags for other actions.